src/main/java/io/xpire/logic/commands/SearchCommand.java

```diff
@@ -30,8 +36,24 @@ public SearchCommand(ContainsKeywordsPredicate predicate) {
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.updateFilteredItemList(this.predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_ITEMS_LISTED_OVERVIEW, model.getFilteredItemList().size()));
+        StringBuilder sb = new StringBuilder(String.format(Messages.MESSAGE_ITEMS_LISTED_OVERVIEW,
+                model.getFilteredItemList().size()));
+        if (model.getFilteredItemList().size() == 0) {
+            predicate.getKeywords().forEach(s -> {
+                if (s.startsWith("#")) {
+                    sb.append(StringUtil.findSimilar(s, model.getAllItemTags().stream()
+                                                             .map(Tag::toString)
+                                                             .collect(Collectors.toSet()), 3));
+                } else {
+                    sb.append(StringUtil.findSimilar(s, model.getAllItemNames().stream()
+                                                              .map(Name::toString)
+                                                              .map(x -> x.split("\\s+"))
+                                                              .flatMap(Arrays::stream)
+                                                              .collect(Collectors.toSet()), 1));
+                }
+            });
+        }
+        return new CommandResult(sb.toString());
```

Comment:
I feel the if else part can be extracted into two methods. Otherwise the SLAP principle breaks hahah.

Reply:
Do you perhaps mean something like this?
```java
//SearchCommand.java
// if starts with "#"
sb.append(StringUtil.findSimilarItemTags(s, model.getAllItemTags()));
// else
sb.append(StringUtil.findSimilarItemNames(s, model.getAllItemNames()));

//StringUtil.java
public static String findSimilarItemTags(String word, Set<Tag> allTags) {
        return StringUtil.findSimilar(word, allTags.stream()
                                                   .map(Tag::toString)
                                                   .collect(Collectors.toSet()), 3);
}

public static String findSimilarItemNames(String word, Set<Name> allNames) {
        return StringUtil.findSimilar(word, allNames.stream()
                                                    .map(Name::toString)
                                                    .map(x -> x.split("\\s+"))
                                                    .flatMap(Arrays::stream)
                                                    .collect(Collectors.toSet()), 1);
}
```

Reply:
Have made the above stated changes for now